ui/models.py

```diff
@@ -66,24 +69,85 @@ def s3_key(self):
             user=self.creator.id, uuid=str(self.s3_subkey), base=basename, ext=extension)
         return newkey
 
+    @property
+    def job_status(self):
+        """
+        Check the encode job status & if not complete, update the status via a query to AWS.
+
+        Returns:
+            str: The video's latest encode job status
+
+        """
+        if self.status in ['Complete', 'Error']:
+            return self.status
+        else:
+            job = get_et_job(self.encode_jobs.latest("created_at").id)
+            if job['Status'] == 'Complete':
+                self.process_transcode_results(job)
+            self.status = job['Status']
+            self.save()
+            return self.status
+
+    def transcode_key(self, preset=None):
+        """
+        Get the S3 key for a video playlist file
+
+        Args:
+            encoding(str): The encoding preset to use (should be one included in settings.ET_PRESET_IDS)
+
+        Returns:
+            str: The S3 key to used for the encoded file.
+        """
+        original_s3_key = self.videofile_set.get(encoding='original').s3_object_key
+        if not preset:
+            return original_s3_key
+        output_template = 'transcoded/{}_{}'
+        basename, _ = os.path.splitext(original_s3_key)
+        return output_template.format(basename, preset)
+
+    def process_transcode_results(self, job):
```

Comment:
I don't think a model is an appropriate place for this method. Usually we'd put these in a method like `api.py`.